content/courses/onchain-development/local-setup.md

```diff
@@ -2,116 +2,219 @@
-You can open `programs/favorites/src/lib.rs` and change `ctx` to `_ctx` or just
-go onto the next step.
+     ```bash
+     cargo add solana-program
+     ```
 
-##### `No license field in package.json` warning
+   - **Manually Update Cargo.toml**: Alternatively, add the following line to
+     your Cargo.toml file:
 
-Open package.json, add `"license": "MIT"` or `"license": "UNLICENSED"` depending
-on preferences
+     ```bash
+     solana-program = "2.0.8"
+     ```
+
+   You can find the latest stable version on the
+   [solana-program crate page](https://crates.io/crates/solana-program).
+
+3. **Re-run Tests**: After upgrading, re-run your tests with the following
+   command:
+
+   ```bash
+   anchor test
+   ```
+
+#### Warning: `unused variable: 'ctx'`
+
+This warning indicates that the `initialize` instruction handler isn’t doing
+anything yet. You can either open `programs/favorites/src/lib.rs` and change
+`ctx` to `_ctx` or proceed to the next step.
+
+#### Warning: `No license field in package.json`
+
+Open `package.json`, add `"license": "MIT"` or `"license": "UNLICENSED"`
+depending on preferences.
+
+#### Keeping Anchor Keys in Sync
+
+The `anchor keys sync` command is used to synchronize the keypairs between the
+local environment and the `Anchor.toml` configuration file. This is particularly
+useful when the local keypair files are not in sync with the deployed program's
+public keys.
+
+#### When to Use anchor keys sync
+
+- **After creating a new keypair**: If you generate a new keypair for your
+  program, run `anchor keys sync` to update the `Anchor.toml` file with the new
+  keypair.
+- **If you've modified keypairs manually**: If you change any keypair files
+  directly, use this command to ensure that `Anchor.toml` reflects those
+  changes.
+- **Before deploying a program**: Running this command ensures that the correct
+  keypair is used during the deployment.
+- **After Cloning a Repository**: If you clone a project that has already been
+  deployed, run `anchor keys sync` to fetch and sync the keypair files based on
+  the program IDs in `Anchor.toml`.
+
+#### Potential Issues if Keys Are Not in Sync
+
+If the keypairs are not synchronized, you may encounter errors when deploying or
+interacting with the onchain program. For example, Anchor may reference an
+incorrect or non-existent keypair, leading to errors such as:
+
+- **Failed transactions**: If the wrong keypair is used, transactions might fail
+  due to invalid signatures or permissions.
+- **Deployment issues**: The program might not deploy successfully if the
+  keypair in `Anchor.toml` does not match the keypair on your machine.
+- **Invalid Program Error:** If the keys are not in sync, you might encounter
+  errors like `Error: Program does not exist` or `Error: Invalid program ID`
+  when interacting with the program via anchor test or other commands.
+
+To avoid these issues, always run `anchor keys sync` after any changes to
+keypairs.
```

Comment:
This is good content, but it doesn't belong in the local setup section. People are not generating new keypairs for the program they just made with `anchor init`. We want to keep this chapter focused on making sure anchor init works, and not cover other topics.

Reply:
Yes deleted this. I will add it to another relevant lesson if applicable.